src/main.ts

```diff
@@ -585,9 +586,10 @@ async function main() {
           );
         }
 
-        testSuite.tests = filterTests(testSuite.tests, {
+        testSuite.tests = await filterTests(testSuite, {
           firstN: cmdObj.firstN,
           pattern: cmdObj.pattern,
+          failing: cmdObj.failing,
```

Comment:
I'm wondering if these command line params should be changed to be something like:
```
filterFirstN
filterPattern
filterFailing
```

Reply:
Ok... I'll make the change in this PR